### PR TITLE
[Dialogue] Part 8: Legacy Lock Service Locks

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Add Dialogue support for high frequency endpoints on the V1 lock service which can still account for as much as 25% of traffic at some stacks. Refresh and unlock are handled in the base PR #4790, this handles lock

**Implementation Description (bullets)**:
- Conjurize `lockAndGetHeldLocks`.

**Testing (What was existing testing like?  What have you done to improve it?)**:
- Added tests that validate that the same underlying service is used (e.g. I locked something in conjure and can't in legacy, vice versa)

**Concerns (what feedback would you like?)**:
Not much

**Where should we start reviewing?**: lock-api.yml

**Priority (whenever / two weeks / yesterday)**: this week

### CODE REVIEW

File: lock-conjure-api/src/main/conjure/lock-api.yml [hunk 1]
```diff
@@ -29,6 +40,12 @@ services:
     package: com.palantir.lock
     base-path: /lk
     endpoints:
+      lockAndGetHeldLocks:
+        http: POST /laghl/{namespace}
```
Comment (gmaretic) [inline lock-conjure-api/src/main/conjure/lock-api.yml:44]: missed an opportunity to have a `/laugh` endpoint

File: timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java [hunk 2]
```diff
@@ -362,6 +365,41 @@ public void directLegacyAndConjureLockServicesInteractCorrectly() throws Interru
...
+        HeldLocksToken token = client.conjureLegacyLockService().lockAndGetHeldLocks(
+                AuthHeader.valueOf("Bearer unused"),
+                client.namespace(),
+                conjureLockRequest)
+                .orElseThrow(() -> new RuntimeException("We should have been able to get the lock"));
+
+        assertThat(client.legacyLockService().lockAndGetHeldLocks(anonymousId, lockRequest))
+                .as("if the conjure impl has taken a lock, the legacy impl mustn't be able to take it")
+                .isNull();
+        assertThat(client.legacyLockService().unlock(token.getLockRefreshToken())).isTrue();
```
Comment (gmaretic) [inline timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java:391]: as("legacy impl can unlock a lock taken by conjure impl")

File: timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java [hunk 3]
```diff
@@ -362,6 +365,41 @@ public void directLegacyAndConjureLockServicesInteractCorrectly() throws Interru
...
+        assertThat(client.legacyLockService().lockAndGetHeldLocks(anonymousId, lockRequest))
+                .as("a lock once unlocked by conjure impl can be acquired by legacy impl")
```
Comment (gmaretic) [inline timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java:393]: not unlocked by conjure impl

Comment (jeremyk-91) [reply timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java:393]: oops 🔪 yeah, `lock taken by conjure impl that was unlocked, can now be acquired by legacy impl`

File: lock-conjure-api/src/main/conjure/lock-api.yml (hunk 1 above)
Comment (jeremyk-91) [reply lock-conjure-api/src/main/conjure/lock-api.yml:44]: I could rename the endpoint to `lockAndUltimatelyGetHeld` 😄

**Pull request merged**